umake/frameworks/ide.py

```diff
@@ -473,3 +474,83 @@ def post_install(self):
                                                      categories=categories))
         if not self.was_in_arduino_group:
             UI.delayed_display(DisplayMessage(_("You need to logout and login again for your installation to work")))
+            
+class BaseNetBeans(umake.frameworks.baseinstaller.BaseInstaller):
+    """The base for all Netbeans installers."""
+    
+    BASE_URL = "http://download.netbeans.org/netbeans/8.0.2/final/"
+
+    def __init__(self, category, version="8.0.2", flavour=""):
+        #add a separator to the string, like -cpp
+        if flavour != "":
+            flavour = '-' + flavour
+        super().__init__(name="Netbeans",
+                         description=_("Netbeans IDE " + version),
+                         category=category, 
+                         only_on_archs=['i386', 'amd64', 'arm'],
+                         download_page=None,
+                         dir_to_decompress_in_tarball="netbeans" + flavour,
+                         desktop_filename="netbeans" + flavour + ".desktop",
```

Comment:
I will update to follow that style, it's the first time that I work only using the source code as documentation 

In UK it's flavour, but I can change it if that makes things more consistent.

I think I misunderstood the meaning of the dir_to_decompress_in_tarball variable. I thought it was the destination folder. Then I will need to change it, as the required dirs are almost all but those with Mac OS and Windows specific libraries.